core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -3112,10 +3110,15 @@ public PhysicalOperation visitSemiJoin(SemiJoinNode node, LocalExecutionPlanCont
                 log.debug("[Semi-join] Dynamic filter: %s", filterId);
                 ImmutableList.Builder<Consumer<Map<DynamicFilterId, Domain>>> collectors = ImmutableList.builder();
                 if (isLocalDynamicFilter) {
-                    collectors.add(context::addLocalDynamicFilters);
+                    LocalDynamicFiltersCollector localDynamicFiltersCollector = context.getDynamicFiltersCollector();
+                    collectors.add(localDynamicFiltersCollector::collectDynamicFilterDomains);
                 }
                 if (isCoordinatorDynamicFilter) {
-                    collectors.add(context.getCoordinatorDynamicFilterDomainsCollector(ImmutableSet.of(filterId)));
+                    TaskContext taskContext = context.getTaskContext();
+                    collectors.add(domains -> taskContext.updateDomains(
+                            domains.entrySet().stream()
+                                    .filter(entry -> filterId.equals(entry.getKey()))
```

Comment:
this is different check than:
```
.filter(entry -> coordinatorDynamicFilters.contains(entry.getKey()))
```

Reply:
That's because `coordinatorDynamicFilters = ImmutableSet.of(filterId)`

Reply:
Why? `getCoordinatorDynamicFilters` has more conditions. Also `isCoordinatorDynamicFilter` might be `false`

Reply:
I'm not sure what you're asking, the code written here is just an inlining of `context.getCoordinatorDynamicFilterDomainsCollector`, it should keep doing whatever it was doing before.

Reply:
I don't see transition from `.filter(entry -> coordinatorDynamicFilters.contains(entry.getKey()))` into `.filter(entry -> filterId.equals(entry.getKey()))`.

Semi Join is similar to normal join. In pipeline mode for REPLICATED semi-join only single task will collect DF. That is governed by `getCoordinatorDynamicFilters` or `isCoordinatorDynamicFilter` local variable here

Reply:
I've now updated the code to make it obvious that there is no logical change happening here